File: backend/core-service/src/main/java/com/tuanhust/coreservice/service/TaskService.java
package com.tuanhust.coreservice.service;

import com.tuanhust.coreservice.request.TaskRequest;
import com.tuanhust.coreservice.response.CheckListResponse;
import com.tuanhust.coreservice.response.CommentResponse;
import com.tuanhust.coreservice.response.TaskDetailResponse;
import com.tuanhust.coreservice.response.TaskResponse;

import java.util.List;

public interface TaskService {
    TaskResponse createTask(String projectId, TaskRequest taskRequest);

    List<TaskResponse> getTaskForProject(String projectId);

    TaskResponse archiveTask(String projectId, String taskId);

    TaskResponse restoreTask(String projectId, String taskId, Double sortOrder);

    void deleteTask(String projectId, String taskId);

    TaskResponse moveTask(String projectId, String taskId, Double sortOrder, String boardColumnId);

    TaskDetailResponse getTask(String projectId, String taskId);

    void updateTask(String projectId, String taskId, TaskRequest taskRequest);

    void updateCompletedTask(String projectId, String taskId, Boolean completed);

    CheckListResponse createCheckList(String projectId, String taskId, String body);

    CheckListResponse updateCheckList(String checkListId, String body, Boolean done);

    void deleteCheckList(String checkListId);

    CommentResponse createComment(String projectId, String taskId, String body);

    void deleteComment(String commentId);

    CommentResponse updateComment(String projectId,String commentId, String body);
}

File: fontend/task-management-fe/src/features/projects/services/TaskService.ts
import {privateApi} from '@config/api.config';
import {TaskResponse} from '@features/task/types';
import {CheckListResponse, CommentResponse, TaskDetailResponse, TaskRequest} from "@features/projects/types/task.types";

export const TaskService = {
    getTasksByProject: async (projectId: string): Promise<TaskResponse[]> => {
        return await privateApi.get(`/project/${projectId}/task`);
    },
    createTask: async (projectId: string, taskRequest: TaskRequest): Promise<TaskResponse> => {
        return await privateApi.post(`/project/${projectId}/task`, taskRequest);
    },
    archiveTask: async (projectId: string, taskId: string): Promise<TaskResponse> => {
        return await privateApi.post(`/project/${projectId}/task/${taskId}/archive`);
    },
    restoreTask: async (projectId: string, taskId: string, sortOrder?: number): Promise<TaskResponse> => {
        return await privateApi.post(
            `/project/${projectId}/task/${taskId}/restore`,
            {},
            {
                params: {
                    sortOrder: sortOrder
                }
            }
        );
    },
    deleteTask: async (projectId: string, taskId: string): Promise<void> => {
        return await privateApi.delete(`/project/${projectId}/task/${taskId}`);
    },
    moveTask: async (projectId: string, taskId: string, data: {
        sortOrder: number,
        boardColumnId: string
    }): Promise<TaskResponse> => {
        return await privateApi.patch(`/project/${projectId}/task/${taskId}/move`, {},
            {
                params: data
            });
    },
    getTask: async (projectId: string, taskId: string): Promise<TaskDetailResponse> => {
        return await privateApi.get(`/project/${projectId}/task/${taskId}`);
    },
    updateTask: async (projectId: string, taskId: string, data: Partial<TaskRequest>): Promise<void> => {
        return await privateApi.put(`/project/${projectId}/task/${taskId}`, data);
    },
    updateDone: async (projectId: string, taskId: string, completed: boolean): Promise<void> => {
        return await privateApi.put(`/project/${projectId}/task/${taskId}/complete`, {},
            {
                params: {
                    completed: completed
                }
            });
    },
    createCheckList: async (projectId: string, taskId: string, body: string): Promise<CheckListResponse> => {
        return await privateApi.post(`/project/${projectId}/task/${taskId}/checkList`, {},
            {
                params: {
                    body: body
                }
            });
    },
    updateCheckList: async (
        projectId: string,
        taskId: string,
        checkListId: string,
        data: { body?: string; done?: boolean }
    ): Promise<CheckListResponse> => {
        return await privateApi.put(
            `/project/${projectId}/task/${taskId}/checkList/${checkListId}`,
            {},
            {params: data}
        );
    },
    deleteCheckList: async (
        projectId: string,
        taskId: string,
        checkListId: string,
    ): Promise<CheckListResponse> => {
        return await privateApi.delete(
            `/project/${projectId}/task/${taskId}/checkList/${checkListId}`
        );
    },
    createComment: async (projectId: string, taskId: string, body: string): Promise<CommentResponse> => {
        return await privateApi.post(`/project/${projectId}/task/${taskId}/comment`, {body: body});
    },
    deleteComment: async (projectId: string, taskId: string, commentId: string): Promise<void> => {
        return await privateApi.delete(`/project/${projectId}/task/${taskId}/comment/${commentId}`);
    },
    updateComment: async (projectId: string, taskId: string, commentId: string, body: string): Promise<CommentResponse> => {
        return await privateApi.put(`/project/${projectId}/task/${taskId}/comment/${commentId}`, {body: body});
    },
};